### PR TITLE
BFD-2157: Convert BFD java keystore from JKS to tool-agnostic formats

### PR DESCRIPTION
**Pull request by malessi**
**JIRA Ticket:**
[BFD-2157](https://jira.cms.gov/browse/BFD-2157)

**User Story or Bug Summary:**

As a BFD engineer, I want the BFD certificate chain to be represented in tool-agnostic formats so that I can better maintain, store, and securely deliver them to bfd-server instances.

---

### What Does This PR Do?

This PR updates all JKS keystores/truststores (including local/`dev` stores) to use PKCS12. Additionally, the established environment keystores have been moved into SSM entirely as secure parameters managed by the `base` Terraservice and have been removed from the repository. All references to the now deprecated JKS stores have been replaced with references to the new PKCS12 stores, as well. 

Note that the stores have not changed with respect to the certificate chain(s) that they store, only the _format_ of these stores have been changed. This means that existing certificates will still work as expected.

This PR has been verified by:

- the `CI - Java` GHA check passing verifying that the ansible tests pass successfully for the `bfd-server` role
- verifying that exporting public certificates from the converted PKCS12 keystores _match_ the existing public certificates exported from the deprecated JKS keystores
- building the `bfd-server` app AMI verifying that it builds properly
- building BFD Server locally and:
  - running it locally verifying that requests using the "unsecured" certificates work as expected
  - running the ITs using test containers and verifying that they pass
- creating a new ephemeral environment, `bfd2157-test` (has since been `destroy`ed), and:
  - `apply`ing `base`, `common`, and `server` and verifying that `server` `apply`s as expected
  - SSHing into the `fhir` instance created and verifying that the keystore is valid using `keytool`
  - running the regression suite against the `bfd2157-test` `fhir` instance and verifying that there were no failures

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

- Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

- If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

  - Does this PR add any new software dependencies?
    - [ ] Yes
    - [x] No
  - Does this PR modify or invalidate any of our security controls?
    - [ ] Yes
    - [x] No
  - Does this PR store or transmit data that was not stored or transmitted before?
    - [ ] Yes
    - [x] No

- If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
  - Do you think this PR requires additional review of its security implications for other reasons?
    - [ ] Yes
    - [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

- N/A

### Submitter Checklist

I have gone through and verified that...:

- [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
- [x] This PR is reasonably limited in scope, to help ensure that:
  1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
  2. There isn't too much of a burden on reviewers.
  3. Any problems it causes have a small "blast radius".
  4. It'll be easier to rollback if that becomes necessary.
- [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
- [x] The data dictionary has been updated with any field mapping changes, if any were made.
- [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
- [x] Reviews are requested from both:
  - At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
  - Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
- [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
  - Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — APPROVED** [review]
THANK YOU! Good to be rid of JKS!

File: ops/terraform/services/base/sensitive.tf [hunk 1]
```diff
@@ -19,6 +19,7 @@ resource "aws_ssm_parameter" "common_sensitive" {
   name      = each.key
   overwrite = true
   type      = "SecureString"
+  tier      = "Intelligent-Tiering"
```
Comment (malessi) [inline ops/terraform/services/base/sensitive.tf:22]: I've added `Intelligent-Tiering` to each `resource` here as the keystore parameter is larger than the maximum supported size of 4 KB for the `Standard` tier of SSM parameters. Fortunately, `Intelligent-Tiering` will automatically create the keystore parameter as an `Advanced` tier parameter, which have a limit of 8 KB. 

`Advanced` SSM parameters have a cost of:

- $0.05 per month per `Advanced` parameter
- $0.05 per 10,000 API interactions

This means, this new parameter (per-environment) will cost about $2 per year to store and likely less than $1 per year to access, which I felt was acceptable. We can discuss if a different approach should be taken here, if necessary.

**Review by brianburton — COMMENTED** [review]
Excellent change.  Sounds as though it was tested thoroughly.

**Review by brianburton — APPROVED** [review]

File: apps/bfd-server/dev/devenv-readme.md [hunk 2]
```diff
@@ -36,29 +36,42 @@ For your convenience, a dev-only-really-don't-use-these-anywhere-else server key
     ```
     $ keytool -genkeypair -alias server -keyalg RSA -keysize 4096 -dname "cn=localhost" -ext "san=dns:localhost,ip:127.0.0.1" -validity 3650 -keypass changeit -keystore bluebutton-server.git/dev/ssl-stores/server-keystore.jks -storepass changeit
     ```
+    1. Do note that _all_ JKS keystores have since been updated to use the more widely accepted standard PKCS12 format. Using `server-keystore.jks` as an example, this conversion was achieved with the following `keytool` command:
+
+        ```
+        keytool -importkeystore -srckeystore ./server-keystore.jks -destkeystore server-keystore.pfx -srcstoretype jks -deststoretype pkcs12 -srcstorepass changeit -deststorepass changeit -srcalias server -destalias server -srckeypass changeit -destkeypass changeit -noprompt
+        ```
+    2. Do note that this document will continue to refer to the (historical) certificate as if it were in JKS format and named `server-keystore.jks`. Any reference to `server-keystore.jks` can be replaced with `server-keystore.pfx`
```
Comment (lsmitchell) [inline apps/bfd-server/dev/devenv-readme.md:45]: Any reason not to just find and replace the instances here?

**Review by lsmitchell — APPROVED** [review]
👍

File: apps/bfd-server/dev/devenv-readme.md (hunk 2 above)
Comment (mjburling) [reply apps/bfd-server/dev/devenv-readme.md:45]: > Any reason not to just find and replace the instances here?

Between Brian, Mitch, and I, we were looking at this very thing.

I realize that the commit history does a fine job by itself recording things for posterity, but with some of the shifts recently and work we're scheduling in the coming months, I agree with Mitch's lighter touch here. I think it's reasonable to continue displaying all of this context:
* we're forced to revisit aspects of BFD's authentication strategy in PI5 with certificate expiration later this year
* we're starting to take a closer look at the documentation strategy in the BFD project

**Review by mjburling — APPROVED** [review]

**Pull request merged**